packages/firestore/src/local/memory_persistence.ts

```diff
@@ -277,9 +277,8 @@ export class MemoryEagerDelegate implements ReferenceDelegate {
             // Since this is the eager delegate and memory persistence,
             // we don't care about the size of documents. We don't track
             // the size of the cache for eager GC.
-            return cache.removeEntry(txn, key).next(() => {});
+            cache.removeEntry(key);
```

Comment:
Consider using RemoteDocumentChangeBuffer here too so that we can make removeEntry private...

Reply:
Done